File: src/DepositfilesSDK/DepositfilesSDK/Interfaces/IRoot.vb
Imports DepositfilesSDK.JSON

Public Interface IRoot

    Function ListFolders() As Task(Of List(Of JSON.JSON_FolderMetadata))

    Function ListFiles() As Task(Of JSON.JSON_List)

    Function CreateNewFolder(FolderName As String) As Task(Of JSON.JSON_FolderMetadata)

    Function SearchFiles(SearchKeyword As String, SearchType As utilitiez.SearchTypeEnum) As Task(Of List(Of JSON_FileMetadata))

End Interface

File: DepositfilesSDK/DepositfilesSDK/Interfaces/IRoot.cs
using DepositfilesSDK.JSON;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace DepositfilesSDK
{
    public interface IRoot
    {

        string RootID { get; }
        Task<List<JSON_FolderMetadata>> ListFolders();
        Task<JSON_List> ListFiles();
        Task<JSON_FolderMetadata> CreateNewFolder(string FolderName);
        Task<List<JSON_FileMetadata>> SearchFiles(string SearchKeyword, Utilitiez.SearchTypeEnum SearchType);

    }
}
